plugins/kotlin-serialization/kotlin-serialization-compiler/src/org/jetbrains/kotlinx/serialization/compiler/backend/ir/SerializableCompanionIrGenerator.kt

```diff
@@ -84,6 +105,70 @@ class SerializableCompanionIrGenerator(
         irSerializableClass.annotations += annotationCtorCall
     }
 
+    override fun generateLazySerializerGetter(methodDescriptor: FunctionDescriptor) {
+        val serializerDescriptor = requireNotNull(
+            findTypeSerializer(
+                serializableDescriptor.module,
+                serializableDescriptor.toSimpleType()
+            )
+        )
+
+        val field = irClass.addField {
+            name = Name.identifier(SerialEntityNames.SERIALIZER_LAZY_DELEGATE_FIELD_NAME)
+            visibility = DescriptorVisibilities.PRIVATE
+            origin = SERIALIZABLE_PLUGIN_ORIGIN
+            isFinal = true
+            isStatic = true
+            type = lazyDescriptor.defaultType.toIrType()
+        }.apply {
+            val lambda = irClass.createSingletonLambda(serializerDescriptor.defaultType.toIrType(), startOffset, endOffset) {
+                val expr = serializerInstance(
+                    this@SerializableCompanionIrGenerator,
+                    serializerDescriptor, serializableDescriptor.module,
+                    serializableDescriptor.defaultType
+                )
+                patchSerializableClassWithMarkerAnnotation(serializerDescriptor)
```

Comment:
`patchSerializableClassWithMarkerAnnotation` can't be placed near `initializer = irClass.factory.createExpressionBod` because its extension for `IrBuilderWithScope`. So it can be located only in `createSingletonLambda` or `contributeFunction` blocks.
Not sure if there is a fundamental difference between them.